### PR TITLE
Zhengqian/add auto scaling

### PR DESCRIPTION
**Pull request by buggithubs**

### CODE REVIEW

File: op-guide/auto-scaling.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+> 根据部署情况选择用户场景设置
+
+> 本次案例默认操作已经做主机互信
+
+### 主机用户场景
+
+#### A 场景
+
+1. 客户集群主机拥有小用户(非root用户,且root用户无法远程登录),小用户拥有sudo权限
```
Comment (disksing) [inline op-guide/auto-scaling.md]: 中英文空格

**Comment by buggithubs** [comment]
@disksing @queenypingcap @liubo0127 @LinuxGit @zhengwanbo

**Comment by disksing** [comment]
全角标点和半角标点太乱了……整理一下吧

File: op-guide/auto-scaling.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
+---
+title: TiDB 弹性伸缩 & 下线方案
+category: deployment
+---
+
+# TiDB 弹性伸缩 & 下线 方案
+
+## 用户场景
+
+> 被操作环境是由 tidb-ansible 部署
```
Comment (andelf) [inline op-guide/auto-scaling.md]: Wrong Style

File: op-guide/auto-scaling.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+> 根据部署情况选择用户场景设置
+
+> 本次案例默认操作已经做主机互信
+
+### 主机用户场景
+
+#### A 场景
+
+1. 客户集群主机拥有小用户(非 root 用户,且 root 用户无法远程登录),小用户拥有 sudo 权限
```
Comment (andelf) [inline op-guide/auto-scaling.md]: wrong style

File: op-guide/auto-scaling.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+-----
+
+## 扩容 & 修复节点
+
+> 扩容与修复单节点方式类似  
+
+> 扩容 PD 与 TiKV TiDB 操作步骤不同, 操作时请分开操作
+
+#### 现有集群
```
Comment (andelf) [inline op-guide/auto-scaling.md]: Never use level-4 title in a single file.
And this is not a title

File: op-guide/auto-scaling.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+> PD * 3 , TiDB * 2 , TiKV * 3
```
Comment (andelf) [inline op-guide/auto-scaling.md]: quote block is only used for notice

File: op-guide/auto-scaling.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+| node2 | 172.16.10.2 | PD2, TiDB2 |
+| node3 | 172.16.10.3 | PD3, Monitor|
+| node4 | 172.16.10.4 | TiKV1 |
+| node5 | 172.16.10.5 | TiKV2 |
+| node6 | 172.16.10.6 | TiKV3 |
+
+
+#### 扩容 tikv & tidb
+
+> 本次新增 tidb3 与 tikv4
```
Comment (andelf) [inline op-guide/auto-scaling.md]: !

File: op-guide/auto-scaling.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+| node1 | 172.16.10.1 | PD1, TiDB1 |
+| node2 | 172.16.10.2 | PD2, TiDB2 |
+| node3 | 172.16.10.3 | PD3, Monitor |
+| **node101** | **172.16.10.101**|**TiDB3** |
+| node4 | 172.16.10.4 | TiKV1 |
+| node5 | 172.16.10.5 | TiKV2 |
+| node6 | 172.16.10.6 | TiKV3 |
+| **node102** | **172.16.10.102**|**TiKV4** |
+
+> 以下为具体操作步骤
```
Comment (andelf) [inline op-guide/auto-scaling.md]: ！

File: op-guide/auto-scaling.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+- 更新 inventory.ini 文件
```
Comment (andelf) [inline op-guide/auto-scaling.md]: use ordered list

File: op-guide/auto-scaling.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+    - 添加 IP 到inventory.ini 
+    - 注意修改用户场景
+- 初始化新增机器
+    - 之前机器不会被初始化，只会操作新机器，需要 root 权限
+    - ansible-playbook bootstrap.yml -D
```
Comment (andelf) [inline op-guide/auto-scaling.md]: use backquote
  
    ``your command``

File: op-guide/auto-scaling.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+    - 不会影响已存在的服务运行  
+    - ansible-playbook deploy.yml -D
+- 滚动重启节点
+    - ansible-playbook rolling_update.yml --tags=tidb -l 172.16.10.101
+    - ansible-playbook rolling_update.yml --tags=tikv -l 172.16.10.102 
+
+
+#### 扩容 PD 节点
+
+> tikv tidb 需要更新配置文件中 pd 参数，所以扩容 PD 节点需要滚动重启所有节点。
```
Comment (andelf) [inline op-guide/auto-scaling.md]: !

File: op-guide/auto-scaling.md [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+    - 手动启动 PD 节点
+        - `{deploy_dir}/scripts/start_pd.sh`
+    - 通过 pd-api 查看是否有相应的节点信息
+        - `curl http://172.16.10.1:2379/pd/api/v1/members` 
+- 滚动重启 tidb & tikv 节点，滚动重启时刷新 tidb & tikv 配置文件
+    - ansible-playbook rolling_update.yml --tags=tidb
+    - ansible-playbook rolling_update.yml --tags=tikv
+
+
+-----
```
Comment (andelf) [inline op-guide/auto-scaling.md]: DO NOT use h-rule

File: op-guide/auto-scaling.md [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+| node3 | 172.16.10.3 | PD3,Monitor | 
+| node104 | 172.16.10.104|PD4 |
+| node101 | 172.16.10.101|TiDB3 |
+| node4 | 172.16.10.4 | TiKV1 |
+| node5 | 172.16.10.5 | TiKV2 |
+| node6 | 172.16.10.6 | TiKV3 |
+| node102 | 172.16.10.102 | TiKV4 |
+
+
+> 以下为具体操作步骤
```
Comment (andelf) [inline op-guide/auto-scaling.md]: !

File: op-guide/auto-scaling.md [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+| node4 | 172.16.10.4 | TiKV1 |
+| node5 | 172.16.10.5 | TiKV2 |
+| **node6** | **172.16.10.6** | **TiKV3** |
+| node102 | 172.16.10.102 | TiKV4 |
+
+
+> 以下为具体操作步骤
+
+- 使用 pd-ctl 工具检查 `replica-schedule-limit` 参数
+    - `./pd-ctl -u "http://172.16.10.1:2379"`
```
Comment (andelf) [inline op-guide/auto-scaling.md]: use code block.

File: op-guide/auto-scaling.md [hunk 14]
```diff
@@ -0,0 +1,260 @@
...
+| node4 | 172.16.10.4 | TiKV1 |
+| node5 | 172.16.10.5 | TiKV2 |
+| node6 | 172.16.10.6 | TiKV3 |
+| node102 | 172.16.10.102 | TiKV4 |
+
+
+> 以下为具体操作步骤
+
+- 使用 pd-ctl 工具下线 PD 节点
+    - `./pd-ctl -u "http://172.16.10.1:2379"`
```
Comment (andelf) [inline op-guide/auto-scaling.md]: use code block

**Comment by buggithubs** [comment]
@andelf PTAL

**Pull request closed without merging**